webroot/src/network/mocks/mock.data.api.ts

```diff
@@ -71,8 +71,23 @@ export class DataApi {
     }
 
     // Get Licensee by ID
-    public getLicensee() {
-        return this.wait(500).then(() => LicenseeSerializer.fromServer(licensees.items[0]));
+    public getLicensee(licenseeId, params: any = {}) {
+        const serverResponse = licensees.items.find((item) => item.providerId === licenseeId);
+        let response;
+
+        if (serverResponse) {
+            response = this.wait(500).then(() => ({
+                licensee: LicenseeSerializer.fromServer(licensees.items[0]),
+                licenseeId,
+                params,
+            }));
+        } else {
+            response = this.wait(500).then(() => {
+                throw new Error('not found');
+            });
+        }
```

Comment:
If the licensee record was found in the local mock data by ID, then the successful mock response is returned. Otherwise an error response is returned.

The `wait(500)` is just a small delay to make sure our loading UI is working.

Reply:
Ah I totally missed that this was a mock of the data API (for testing, I assume). Thanks for the explanation!

Could we add a brief comment on the intent of `this.wait(500)`?

Reply:
You probably have good reason for the naming, but just curious: any reason to not name it something like `local.data.api.ts`? My first instinct was that this was for testing (once I noticed it was a "mock" file).

Reply:
We have some other scenarios where we have actual local APIs (where a backend stack is running on our dev machines). It's actually more unusual for us to have "mock" endpoints in the backend - and I think there is an upcoming ticket to remove those. So this frontend mock API will be the only mock API soon.

Also, I expanded the comment up at the definition of `wait` to give a clearer example of how it's used.